### PR TITLE
MINOR: Add missing configs for resilience settings

### PR DESCRIPTION
**Pull request by guozhangwang**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@bbejeck @vvcephei

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -519,9 +529,10 @@ <h4><a class="toc-backref" href="#id14">state.dir</a><a class="headerlink" href=
...
-          are defined by specifying parameters in a <code class="docutils literal"><span class="pre">StreamsConfig</span></code> instance.</p>
+        <h3><a class="toc-backref" href="#id16">Kafka consumers, producer and admin client configuration parameters</a><a class="headerlink" href="#kafka-consumers-and-producer-configuration-parameters" title="Permalink to this headline"></a></h3>
+        <p>You can specify parameters for the Kafka <a class="reference external" href="../../../javadoc/org/apache/kafka/clients/consumer/package-summary.html">consumers</a>, <a class="reference external" href="../../../javadoc/org/apache/kafka/clients/producer/package-summary.html">producers</a>,
+            and <a class="reference external" href="../../../javadoc/org/apache/kafka/kafka/clients/admin/package-summary.html">admin client</a> that are used internally.
+            The consumer, producer and admin client settings are defined by specifying parameters in a <code class="docutils literal"><span class="pre">StreamsConfig</span></code> instance.</p>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: Nit: insert Oxford comma (multiple times in both lines)

File: docs/streams/developer-guide/config-streams.html [hunk 2]
```diff
@@ -533,18 +544,21 @@ <h3><a class="toc-backref" href="#id16">Kafka consumers and producer configurati
...
+              <code class="docutils literal"><span class="pre">receive.buffer.bytes</span></code> are used to configure TCP buffers; <code class="docutils literal"><span class="pre">request.timeout.ms</span></code> and <code class="docutils literal"><span class="pre">retry.backoff.ms</span></code> control retries for client request;
+              <code class="docutils literal"><span class="pre">retries</span></code> are used to configure how many retries are allowed when handling retriable errors from broker request responses.
+              You can avoid duplicate names by prefix parameter names with <code class="docutils literal"><span class="pre">consumer.</span></code>, <code class="docutils literal"><span class="pre">producer.</span></code>, or <code class="docutils literal"><span class="pre">admin.</span></code> (e.g., <code class="docutils literal"><span class="pre">consumer.send.buffer.bytes</span></code> and <code class="docutils literal"><span class="pre">producer.send.buffer.bytes</span></code>).</p>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:544]: Should we mention different prefix for different consumers here (not sure)?

File: docs/streams/developer-guide/config-streams.html [hunk 3]
```diff
@@ -533,18 +544,21 @@ <h3><a class="toc-backref" href="#id16">Kafka consumers and producer configurati
...
           <div class="highlight-java"><div class="highlight"><pre><span></span><span class="n">Properties</span> <span class="n">streamsSettings</span> <span class="o">=</span> <span class="k">new</span> <span class="n">Properties</span><span class="o">();</span>
-<span class="c1">// same value for consumer and producer</span>
+<span class="c1">// same value for consumer, producer, and admin client</span>
 <span class="n">streamsSettings</span><span class="o">.</span><span class="na">put</span><span class="o">(</span><span class="s">&quot;PARAMETER_NAME&quot;</span><span class="o">,</span> <span class="s">&quot;value&quot;</span><span class="o">);</span>
 <span class="c1">// different values for consumer and producer</span>
 <span class="n">streamsSettings</span><span class="o">.</span><span class="na">put</span><span class="o">(</span><span class="s">&quot;consumer.PARAMETER_NAME&quot;</span><span class="o">,</span> <span class="s">&quot;consumer-value&quot;</span><span class="o">);</span>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:549]: Should we tackle KIP-267 here, too?

**Review by mjsax — COMMENTED** [review]
Just some nits. Maybe we want a separate PR for KIP-276 anyway.

File: docs/streams/developer-guide/config-streams.html (hunk 2 above)
Comment (bbejeck) [reply docs/streams/developer-guide/config-streams.html:544]: +1 for mentioning the prefix for different consumers here.  IMHO using a separate value for `num.poll.records` for the restore consumer fits into this topic.

**Review by bbejeck — APPROVED** [review]
Just one minor comment but otherwise LGMT

File: docs/streams/developer-guide/config-streams.html (hunk 2 above)
Comment (guozhangwang) [reply docs/streams/developer-guide/config-streams.html:544]: It is already mentioned in line 549 below: `<p>You could further separate consumer configuration by adding different prefixes:</p>`

File: docs/streams/developer-guide/config-streams.html (hunk 3 above)
Comment (guozhangwang) [reply docs/streams/developer-guide/config-streams.html:549]: Ditto.

**Review by mjsax — APPROVED** [review]

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `1.1`

**Pull request merged**